### PR TITLE
[0.7.0] Add auto scaling threshold line to application metrics

### LINKED ISSUE #787: Persistence on Metrics
# Location

- [x] Browser
- [ ] CLI
- [ ] API

# Requirements
On every re-deploy, the metrics tab resets and shows data only from the time of re-deployment. It should show all time data across multiple webhook triggers/updates.
# Additional Details

### PR DESCRIPTION
**Pull request by jnfrati**
## Pull request type

Please check the type of change your PR introduces:

- [ ] Bugfix
- [x] Feature
- [ ] Other (please describe):

## Pull request checklist

Please check if your PR fulfills the following requirements:

- [ ] If it's a backend change, tests for the changes have been added and `go test ./...` runs successfully from the root folder.
- [ ] If it's a frontend change, Prettier has been run
- [ ] Docs have been reviewed and added / updated if needed

## What is the current behavior?

Currently, horizontal pod autoscaling is something difficult to visualize under the metrics tab. There's a lack of history as well as some additional info on when the autoscaling is going to happen

Issue Number: #898 
Closes #787 

## What is the new behavior?

As referenced on the issue, this PR will cover the first graph that will show a threshold that announces the next step before k8s will autoscale the pods as well as new Prometheus metrics that will cover the lack of historical data on dead pods.

## Technical Spec/Implementation Notes

A quick reference on how the graph will look like
image

The new prometheus queries will be handled by regex expressions that depends on the selected controller to get all the pods that have stored data on prometheus.

### CODE REVIEW

**Comment by abelanger5** [comment]
Before this can be merged, the backend implementations should query for metrics named `kube_horizontal_pod_autoscaler...` since this metric was renamed in a recent version of `kube-state-metrics`. It should default to using this metric if it exists, since this also likely means the `kube-state-metrics` chart was upgraded.

File: dashboard/src/main/home/cluster-dashboard/expanded-chart/metrics/AreaChart.tsx [hunk 1]
```diff
@@ -59,243 +54,353 @@ const formats: { [range: string]: (date: Date) => string } = {
...
 };
 
-export default withTooltip<AreaProps, TooltipData>(
-  ({
-    data,
-    resolution,
-    width,
-    height,
-    margin = { top: 0, right: 0, bottom: 0, left: 0 },
+// export default withTooltip<AreaProps, TooltipData>();
```
Comment (abelanger5) [inline dashboard/src/main/home/cluster-dashboard/expanded-chart/metrics/AreaChart.tsx:76]: Remove commented line

File: dashboard/src/main/home/cluster-dashboard/expanded-chart/metrics/MetricNormalizer.ts [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  NormalizedMetricsData,
+  MetricsMemoryDataResponse,
+  MetricsCPUDataResponse,
+  MetricsNetworkDataResponse,
+  MetricsNGINXErrorsDataResponse,
+  AvailableMetrics,
+  MetricsHpaReplicasDataResponse,
+} from "./types";
+
+export class MetricNormalizer {
```
Comment (abelanger5) [inline dashboard/src/main/home/cluster-dashboard/expanded-chart/metrics/MetricNormalizer.ts:12]: I am unsure what this object does. Could you add a comment?

**Review by abelanger5 — CHANGES_REQUESTED** [review]
Few tiny comments @jnfrati, otherwise this is ready to be merged 👍

**Review by abelanger5 — APPROVED** [review]

**Comment by jusrhee** [comment]
@jnfrati turned on autoscaling for a service that didn't have autoscaling enabled initially and got this:
image

Is there a clean way to handle this edge case?

**Pull request merged**